QModManager/Patching/QModFactory.cs

```diff
@@ -0,0 +1,214 @@
+﻿namespace QModManager.Patching
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Reflection;
+    using Oculus.Newtonsoft.Json;
+    using QModManager.API.ModLoading;
+    using QModManager.DataStructures;
+    using QModManager.Utility;
+
+    internal class QModFactory
+    {
+        internal PairedList<QMod, ModStatus> BuildModLoadingList(string qmodsDirectory)
+        {
+            if (!Directory.Exists(qmodsDirectory))
+            {
+                Logger.Info("QMods directory was not found! Creating...");
+                Directory.CreateDirectory(qmodsDirectory);
+
+                return new PairedList<QMod, ModStatus>(0);
+            }
+
+            string[] subDirectories = Directory.GetDirectories(qmodsDirectory);
+            var modSorter = new SortedTree<string, QMod>(subDirectories.Length);
+            var earlyErrors = new PairedList<QMod, ModStatus>(subDirectories.Length);
+
+            foreach (string subDir in subDirectories)
+            {
+                string[] dllFiles = Directory.GetFiles(subDir, "*.dll", SearchOption.TopDirectoryOnly);
+
+                if (dllFiles.Length < 1)
+                    continue;
+
+                QMod mod = FromDll(subDir, dllFiles);
```

Comment:
@AlexejheroYTB I never actually got this working.  
If there is any chance for QMM 3.0 to move forward, we need to rip out the DLL based loading and continue with `mod.json` and/or some other manifest file.

Everything else is still an improvement to mod loading, especially with the 3 patch methods, but this DLL stuff needs to go.

Same goes for SMLHelper.  We will need to re-think (and re-do) the integration with proper consideration for backwards compatibility.